{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "0bdc0a1c-b6a3-455c-b3b4-f42add532090",
   "metadata": {},
   "source": [
    "# Only required on your first run"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "398f35da",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting git+git://github.com/MatthewHiggins2017/bioconda-PrimedRPA.git\n",
      "  Cloning git://github.com/MatthewHiggins2017/bioconda-PrimedRPA.git to /tmp/pip-req-build-6izb9_rc\n",
      "  Running command git clone -q git://github.com/MatthewHiggins2017/bioconda-PrimedRPA.git /tmp/pip-req-build-6izb9_rc\n",
      "  Resolved git://github.com/MatthewHiggins2017/bioconda-PrimedRPA.git to commit 9ac6d7ac1b8503a1aaf27d80d1e6160777c78583\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install git+git://github.com/MatthewHiggins2017/bioconda-PrimedRPA.git"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b9ec3e48-eaf0-40dd-b905-f73955d84379",
   "metadata": {},
   "source": [
    "# Your sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "96532a4e-50f2-4928-940d-94349f5737e3",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Seq\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "3457165e-f230-4488-9270-1eec4549ef8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# The name of the fasta file with your target genome\n",
    "genomeFile = \"sequence.fasta\"\n",
    "\n",
    "#read the genome\n",
    "genome = SeqIO.read(genomeFile, \"fasta\")\n",
    "\n",
    "#PCR Primers, if known\n",
    "forwardPCR = Seq.Seq(\"GACCCCAAAATCAGCGAAAT\")\n",
    "reversePCR = Seq.Seq(\"TCTGGTTACTGCCAGTTGAATCTG\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01e30541-a766-4a03-8c37-f48a06e25dcc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "29721baf-008a-4a71-a580-e0a3c40c8789",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('GACCCCAAAATCAGCGAAAT')"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "forwardPCR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "ed264a15-8b29-4c78-a37b-57f031ee2073",
   "metadata": {},
   "outputs": [],
   "source": [
    "probeRegion = Seq.Seq(\"GCCAAGCAGAACATCCAACTGATCA\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "2fe1fa94-3e90-4875-8004-fedde3ee5cf9",
   "metadata": {},
   "outputs": [],
   "source": [
    "padding = 100\n",
    "start = genome.seq.find(forwardPCR)-padding\n",
    "end = genome.seq.find(reversePCR.reverse_complement())+padding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fdbb2c4e-6717-40c1-918f-dabdb497611b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20ac0ec2-76d4-4926-9432-f80574431052",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Please enter the path, from your current working directory, to the input fasta file:\n",
    "fastaFile = \"trimmed.fasta\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "af9546fd-726c-4558-8726-c35760f95306",
   "metadata": {},
   "source": [
    "# Input Parameters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c8912878-c3a1-42dd-b267-32100649498b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Parameters:\n",
    "#This parameters file will guide the PrimedRPA-based primer and probe design process. Please follow the instructions outlined below:\n",
    "\n",
    "#Please define the reference name for this PrimedRPA run:\n",
    "name = \"N1_Aug5\"\n",
    "\n",
    "#Please indicate if you would like to use a previously generated Alignment File: [NO or File path]\n",
    "preAligned = \"NO\"\n",
    "\n",
    "#Please indicate if you would like to use the previously generated Binding Sites: [NO or File path]\n",
    "knownBindingSites = \"NO\"\n",
    "\n",
    "#Please enter the path, from your current working directory, to the input fasta file:\n",
    "fastaFile = \"trimmed.fasta\"\n",
    "\n",
    "#Please classify the contents of the input fasta file as one of the following options: [SS, MS, AMS]. Whereby:\n",
    "# SS = Single sequence\n",
    "# MS = Multiple unaligned sequences\n",
    "# MAS = Multiple aligned sequences\n",
    "fastaType = \"SS\"\n",
    "\n",
    "#If multiple sequences are present in the input fasta file (Classification of MS or MAS), please indicate below the\n",
    "#percentage identity required for the primers and probes target binding sites:\n",
    "identityRequirement = 99\n",
    "\n",
    "#Please indicate if a primer identity anchor is required. [NO or length of anchor]\n",
    "identityAnchor = \"NO\"\n",
    "\n",
    "#Desired primer length (This can be a range: 28-32 or fixed value: 32):\n",
    "primerLength = \"28-32\"\n",
    "\n",
    "#Please state if you require a probe to be designed and if so what type [NO,EXO,NFO]\n",
    "probeType = \"EXO\"\n",
    "\n",
    "#Desired probe length (This can be a range: 45-50 or fixed value: 50):\n",
    "probeLength = 34\n",
    "\n",
    "#Below please define your max amplicon length.\n",
    "maxAmpliconLength = 150\n",
    "\n",
    "#Below please state the repeat nucleotide cut-off in bp (e.g. 5bp will exclude sequences containing GGGGG).\n",
    "repeatCutoff = 5\n",
    "\n",
    "#Below please insert the minimum percentage GC content for primer/probe:\n",
    "minGCContent = 40\n",
    "\n",
    "#Below please insert the maximum percentage GC content for primer/probe:\n",
    "maxGCContent = 60\n",
    "\n",
    "#Below please indicate the percentage match tolerance for primer-probe dimerisation and secondary structure formation:\n",
    "matchTolerance = 60\n",
    "\n",
    "#Please enter [No or Path to Background file] below to identify if you want to perform a background DNA binding check:\n",
    "backgroundCheck = \"NO\"\n",
    "\n",
    "#Below please insert the percentage background cross reactivity threshold:\n",
    "backgroundThreshold = 65\n",
    "\n",
    "#Below please indicate if you would like to implement a Background Hard Fail Filter [NO,YES]:\n",
    "bgHardFail = \"NO\"\n",
    "\n",
    "#Please define the maximum number of sets you would like to identify:\n",
    "numPrimerSets = 20\n",
    "\n",
    "#Please define the number of threads available:\n",
    "numThreads = 8\n",
    "\n",
    "#Blastn Cross Reactivity Search Settings [Basic or Advanced or Fast]\n",
    "xReactivity = \"Fast\"\n",
    "\n",
    "#Blastn Evalue\n",
    "blastEval = 1000\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6f0f2c65-e28c-4151-a8fd-f23e45b03637",
   "metadata": {},
   "source": [
    "Write the parameter file and run PrimedRPA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5e501077",
   "metadata": {},
   "outputs": [],
   "source": [
    "tFile = open(\"Parameters_template.txt\")\n",
    "template = tFile.read()\n",
    "tFile.close()\n",
    "\n",
    "pFile = open(\"PrimedRPA_Parameters.txt\", 'w')\n",
    "print(template.format(name, preAligned, knownBindingSites, fastaFile, fastaType, identityRequirement, identityAnchor, primerLength, probeType, probeLength, maxAmpliconLength, repeatCutoff, minGCContent, maxGCContent, matchTolerance, backgroundCheck, backgroundThreshold, bgHardFail, numPrimerSets, numThreads, xReactivity, blastEval), file=pFile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5c0b3ec3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "\n",
      "-------------------------------------------\n",
      "----------------PrimedRPA------------------\n",
      "-----Finding RPA Primer and Probe Sets-----\n",
      "-------------Higgins M et al.--------------\n",
      "-------------------------------------------\n",
      "\n",
      "\n",
      "Parameters File Could Not Be Opened\n",
      "Check File Path.\n",
      "Please run PrimedRPA --help to see valid options\n"
     ]
    }
   ],
   "source": [
    "!PrimedRPA PrimedRPA_Parameters.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "02693a28-586f-43db-8b3e-f0e67a3e104e",
   "metadata": {},
   "source": [
    "# Analysis and Picking a good primer set\n",
    "These steps can be done in a spreadsheet or using the scripts below.\n",
    "## Steps:\n",
    "- Ensure the target region is part of the amplicon\n",
    "- Minimize the targe-primer overlap\n",
    "- Minimize amplicon length\n",
    "- Minimize dimerization score\n",
    "- Optimize GC content\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "66e79d51-ef0f-4482-b91d-2d555859c41e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib as mpl \n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fd1dc5ff-61ba-45c0-8aee-4821c8e70d5b",
   "metadata": {},
   "source": [
    "Read the data in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "847ad929-6e0d-44fe-9950-2b19368df7e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "alignmentData = pd.read_csv(f\"{name}_Alignment_Summary.csv\")\n",
    "primerSets = pd.read_csv(f\"{name}_Output_Sets.csv\")\n",
    "bindingSites = pd.read_csv(f\"{name}_PrimedRPA_Oligo_Binding_Sites.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "494444ff-38fe-46ed-84fc-222f0f1b39f2",
   "metadata": {},
   "source": [
    "Look at the primer sets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "c36c210d-d37a-4b47-a62e-c888ad79220b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Forward Primer (FP)</th>\n",
       "      <th>FP GC%</th>\n",
       "      <th>FP Binding Start Site</th>\n",
       "      <th>Reverse Primer (RP)</th>\n",
       "      <th>RP GC%</th>\n",
       "      <th>RP Binding Start Site</th>\n",
       "      <th>Amplicon Size</th>\n",
       "      <th>Max Dimerisation Percentage Score</th>\n",
       "      <th>Max Dimerisation String</th>\n",
       "      <th>Forward Primer Length</th>\n",
       "      <th>Reverse Primer Length</th>\n",
       "      <th>Minimum Primer 3' Identity Anchor</th>\n",
       "      <th>Probe (P)</th>\n",
       "      <th>Probe GC%</th>\n",
       "      <th>Probe Binding Start Site</th>\n",
       "      <th>Probe Length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TCTGATAATGGACCCCAAAATCAGCGAAATGC</td>\n",
       "      <td>43.750</td>\n",
       "      <td>77.0</td>\n",
       "      <td>CCACTGCGTTCTCCATTCTGGTTACTGCCAGT</td>\n",
       "      <td>53.125</td>\n",
       "      <td>143.0</td>\n",
       "      <td>98.0</td>\n",
       "      <td>37.500</td>\n",
       "      <td>TCTGATAATGGACCCCAAAATCAGCGAAATGC          ...</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA</td>\n",
       "      <td>52.941176</td>\n",
       "      <td>109.0</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TCTGATAATGGACCCCAAAATCAGCGAAATGC</td>\n",
       "      <td>43.750</td>\n",
       "      <td>77.0</td>\n",
       "      <td>CCCACTGCGTTCTCCATTCTGGTTACTGCCAG</td>\n",
       "      <td>56.250</td>\n",
       "      <td>144.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>37.500</td>\n",
       "      <td>TCTGATAATGGACCCCAAAATCAGCGAAATGC          ...</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA</td>\n",
       "      <td>52.941176</td>\n",
       "      <td>109.0</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TCTGATAATGGACCCCAAAATCAGCGAAATGC</td>\n",
       "      <td>43.750</td>\n",
       "      <td>77.0</td>\n",
       "      <td>CCCCACTGCGTTCTCCATTCTGGTTACTGCCA</td>\n",
       "      <td>56.250</td>\n",
       "      <td>145.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>37.500</td>\n",
       "      <td>TCTGATAATGGACCCCAAAATCAGCGAAATGC          ...</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA</td>\n",
       "      <td>52.941176</td>\n",
       "      <td>109.0</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TCTGATAATGGACCCCAAAATCAGCGAAATGC</td>\n",
       "      <td>43.750</td>\n",
       "      <td>77.0</td>\n",
       "      <td>GCCCCACTGCGTTCTCCATTCTGGTTACTGCC</td>\n",
       "      <td>59.375</td>\n",
       "      <td>146.0</td>\n",
       "      <td>101.0</td>\n",
       "      <td>40.625</td>\n",
       "      <td>TCTGATAATGGACCCCAAAATCAGCGAAATGC          ...</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA</td>\n",
       "      <td>52.941176</td>\n",
       "      <td>109.0</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TCTGATAATGGACCCCAAAATCAGCGAAATGC</td>\n",
       "      <td>43.750</td>\n",
       "      <td>77.0</td>\n",
       "      <td>CGCCCCACTGCGTTCTCCATTCTGGTTACTGC</td>\n",
       "      <td>59.375</td>\n",
       "      <td>147.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>40.625</td>\n",
       "      <td>TCTGATAATGGACCCCAAAATCAGCGAAATGC          ...</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA</td>\n",
       "      <td>52.941176</td>\n",
       "      <td>109.0</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>AATGTCTGATAATGGACCCCAAAATCAGCGAA</td>\n",
       "      <td>40.625</td>\n",
       "      <td>73.0</td>\n",
       "      <td>CGTTGTTTTGATCGCGCCCCACTGCGTTCTCC</td>\n",
       "      <td>59.375</td>\n",
       "      <td>161.0</td>\n",
       "      <td>120.0</td>\n",
       "      <td>37.500</td>\n",
       "      <td>AATGTCTGATAATGGACCCCAAAATCAGCGAA              ...</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA</td>\n",
       "      <td>52.941176</td>\n",
       "      <td>109.0</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>AATGTCTGATAATGGACCCCAAAATCAGCGAA</td>\n",
       "      <td>40.625</td>\n",
       "      <td>73.0</td>\n",
       "      <td>ACGTTGTTTTGATCGCGCCCCACTGCGTTCTC</td>\n",
       "      <td>56.250</td>\n",
       "      <td>162.0</td>\n",
       "      <td>121.0</td>\n",
       "      <td>37.500</td>\n",
       "      <td>AATGTCTGATAATGGACCCCAAAATCAGCGAA              ...</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA</td>\n",
       "      <td>52.941176</td>\n",
       "      <td>109.0</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>AATGTCTGATAATGGACCCCAAAATCAGCGAA</td>\n",
       "      <td>40.625</td>\n",
       "      <td>73.0</td>\n",
       "      <td>GACGTTGTTTTGATCGCGCCCCACTGCGTTCT</td>\n",
       "      <td>56.250</td>\n",
       "      <td>163.0</td>\n",
       "      <td>122.0</td>\n",
       "      <td>34.375</td>\n",
       "      <td>AATGTCTGATAATGGACCCCAAAATCAGCGAA              ...</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA</td>\n",
       "      <td>52.941176</td>\n",
       "      <td>109.0</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>AATGTCTGATAATGGACCCCAAAATCAGCGAA</td>\n",
       "      <td>40.625</td>\n",
       "      <td>73.0</td>\n",
       "      <td>CGACGTTGTTTTGATCGCGCCCCACTGCGTTC</td>\n",
       "      <td>59.375</td>\n",
       "      <td>164.0</td>\n",
       "      <td>123.0</td>\n",
       "      <td>31.250</td>\n",
       "      <td>AATGTCTGATAATGGACCCCAAAATCAGCGAA              ...</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA</td>\n",
       "      <td>52.941176</td>\n",
       "      <td>109.0</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>AATGTCTGATAATGGACCCCAAAATCAGCGAA</td>\n",
       "      <td>40.625</td>\n",
       "      <td>73.0</td>\n",
       "      <td>CCGACGTTGTTTTGATCGCGCCCCACTGCGTT</td>\n",
       "      <td>59.375</td>\n",
       "      <td>165.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>31.250</td>\n",
       "      <td>AATGTCTGATAATGGACCCCAAAATCAGCGAA              ...</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA</td>\n",
       "      <td>52.941176</td>\n",
       "      <td>109.0</td>\n",
       "      <td>34.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>148 rows Ã— 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Forward Primer (FP)  FP GC%  FP Binding Start Site  \\\n",
       "0    TCTGATAATGGACCCCAAAATCAGCGAAATGC  43.750                   77.0   \n",
       "1    TCTGATAATGGACCCCAAAATCAGCGAAATGC  43.750                   77.0   \n",
       "2    TCTGATAATGGACCCCAAAATCAGCGAAATGC  43.750                   77.0   \n",
       "3    TCTGATAATGGACCCCAAAATCAGCGAAATGC  43.750                   77.0   \n",
       "4    TCTGATAATGGACCCCAAAATCAGCGAAATGC  43.750                   77.0   \n",
       "..                                ...     ...                    ...   \n",
       "143  AATGTCTGATAATGGACCCCAAAATCAGCGAA  40.625                   73.0   \n",
       "144  AATGTCTGATAATGGACCCCAAAATCAGCGAA  40.625                   73.0   \n",
       "145  AATGTCTGATAATGGACCCCAAAATCAGCGAA  40.625                   73.0   \n",
       "146  AATGTCTGATAATGGACCCCAAAATCAGCGAA  40.625                   73.0   \n",
       "147  AATGTCTGATAATGGACCCCAAAATCAGCGAA  40.625                   73.0   \n",
       "\n",
       "                  Reverse Primer (RP)  RP GC%  RP Binding Start Site  \\\n",
       "0    CCACTGCGTTCTCCATTCTGGTTACTGCCAGT  53.125                  143.0   \n",
       "1    CCCACTGCGTTCTCCATTCTGGTTACTGCCAG  56.250                  144.0   \n",
       "2    CCCCACTGCGTTCTCCATTCTGGTTACTGCCA  56.250                  145.0   \n",
       "3    GCCCCACTGCGTTCTCCATTCTGGTTACTGCC  59.375                  146.0   \n",
       "4    CGCCCCACTGCGTTCTCCATTCTGGTTACTGC  59.375                  147.0   \n",
       "..                                ...     ...                    ...   \n",
       "143  CGTTGTTTTGATCGCGCCCCACTGCGTTCTCC  59.375                  161.0   \n",
       "144  ACGTTGTTTTGATCGCGCCCCACTGCGTTCTC  56.250                  162.0   \n",
       "145  GACGTTGTTTTGATCGCGCCCCACTGCGTTCT  56.250                  163.0   \n",
       "146  CGACGTTGTTTTGATCGCGCCCCACTGCGTTC  59.375                  164.0   \n",
       "147  CCGACGTTGTTTTGATCGCGCCCCACTGCGTT  59.375                  165.0   \n",
       "\n",
       "     Amplicon Size  Max Dimerisation Percentage Score  \\\n",
       "0             98.0                             37.500   \n",
       "1             99.0                             37.500   \n",
       "2            100.0                             37.500   \n",
       "3            101.0                             40.625   \n",
       "4            102.0                             40.625   \n",
       "..             ...                                ...   \n",
       "143          120.0                             37.500   \n",
       "144          121.0                             37.500   \n",
       "145          122.0                             34.375   \n",
       "146          123.0                             31.250   \n",
       "147          124.0                             31.250   \n",
       "\n",
       "                               Max Dimerisation String  Forward Primer Length  \\\n",
       "0        TCTGATAATGGACCCCAAAATCAGCGAAATGC          ...                   32.0   \n",
       "1        TCTGATAATGGACCCCAAAATCAGCGAAATGC          ...                   32.0   \n",
       "2        TCTGATAATGGACCCCAAAATCAGCGAAATGC          ...                   32.0   \n",
       "3        TCTGATAATGGACCCCAAAATCAGCGAAATGC          ...                   32.0   \n",
       "4        TCTGATAATGGACCCCAAAATCAGCGAAATGC          ...                   32.0   \n",
       "..                                                 ...                    ...   \n",
       "143  AATGTCTGATAATGGACCCCAAAATCAGCGAA              ...                   32.0   \n",
       "144  AATGTCTGATAATGGACCCCAAAATCAGCGAA              ...                   32.0   \n",
       "145  AATGTCTGATAATGGACCCCAAAATCAGCGAA              ...                   32.0   \n",
       "146  AATGTCTGATAATGGACCCCAAAATCAGCGAA              ...                   32.0   \n",
       "147  AATGTCTGATAATGGACCCCAAAATCAGCGAA              ...                   32.0   \n",
       "\n",
       "     Reverse Primer Length  Minimum Primer 3' Identity Anchor  \\\n",
       "0                     32.0                               32.0   \n",
       "1                     32.0                               32.0   \n",
       "2                     32.0                               32.0   \n",
       "3                     32.0                               32.0   \n",
       "4                     32.0                               32.0   \n",
       "..                     ...                                ...   \n",
       "143                   32.0                               32.0   \n",
       "144                   32.0                               32.0   \n",
       "145                   32.0                               32.0   \n",
       "146                   32.0                               32.0   \n",
       "147                   32.0                               32.0   \n",
       "\n",
       "                              Probe (P)  Probe GC%  Probe Binding Start Site  \\\n",
       "0    ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA  52.941176                     109.0   \n",
       "1    ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA  52.941176                     109.0   \n",
       "2    ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA  52.941176                     109.0   \n",
       "3    ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA  52.941176                     109.0   \n",
       "4    ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA  52.941176                     109.0   \n",
       "..                                  ...        ...                       ...   \n",
       "143  ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA  52.941176                     109.0   \n",
       "144  ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA  52.941176                     109.0   \n",
       "145  ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA  52.941176                     109.0   \n",
       "146  ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA  52.941176                     109.0   \n",
       "147  ACCCCGCATTACGTTTGGTGGACCCTCAGATTCA  52.941176                     109.0   \n",
       "\n",
       "     Probe Length  \n",
       "0            34.0  \n",
       "1            34.0  \n",
       "2            34.0  \n",
       "3            34.0  \n",
       "4            34.0  \n",
       "..            ...  \n",
       "143          34.0  \n",
       "144          34.0  \n",
       "145          34.0  \n",
       "146          34.0  \n",
       "147          34.0  \n",
       "\n",
       "[148 rows x 16 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "primerSets"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
